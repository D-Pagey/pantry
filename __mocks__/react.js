/* eslint-disable global-require */
const reactActual = jest.requireActual('react');

module.exports = {
    ...reactActual,
    // @TODO: remove when warnings are fixed: https://github.com/kentcdodds/react-testing-library/issues/281#issuecomment-465626540
    useState: (initialState) => {
        const [state, setState] = reactActual.useState(initialState);
        return [
            state,
            (value) => {
                require('react-dom/test-utils').act(() => {
                    setState(value);
                });
            }
        ];
    }
};
