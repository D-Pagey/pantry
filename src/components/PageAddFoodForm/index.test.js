import React from 'react';
import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import selectEvent from 'react-select-event';
import { PageAddFoodForm } from '.';
import { Categories } from '../../fixtures/categories';

jest.mock('uuid', () => ({
    v4: () => '5'
}));

const context = {
    categories: Categories,
    updateCategories: () => {},
    updateFridge: () => {}
};

describe('PageAddFoodForm component', () => {
    it('should render', () => {
        const { container } = render(<PageAddFoodForm />, context);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call updateCategories with correct values', async () => {
        const { colour, id, name } = Categories[0];
        const updateCategories = jest.fn();
        const foodName = 'Chicken';
        const { getByTestId, getByLabelText } = render(<PageAddFoodForm />, { ...context, updateCategories });

        await selectEvent.select(getByLabelText('What categories of food?'), [name]);
        await userEvent.type(getByTestId('addFoodInput'), foodName);

        userEvent.click(getByTestId('addFoodFormSubmit'));

        await waitFor(() =>
            expect(updateCategories).toHaveBeenCalledWith([
                {
                    colour,
                    id,
                    label: name,
                    name,
                    value: name
                }
            ])
        );
    });

    it('should call updateFridge with correct values', async () => {
        const { id, name } = Categories[0];
        const updateFridge = jest.fn();
        const foodName = 'Chicken';
        const { getByTestId, getByLabelText, queryAllByText } = render(<PageAddFoodForm />, { ...context, updateFridge });

        await selectEvent.select(getByLabelText('What categories of food?'), [name]);
        await userEvent.type(getByTestId('addFoodInput'), foodName);
        userEvent.click(queryAllByText('Up')[0]);
        userEvent.click(getByTestId('singleSelectButton0'));
        
        userEvent.click(getByTestId('addFoodFormSubmit'));

        await waitFor(() => expect(updateFridge).toHaveBeenCalledWith({
            categories: [id],
            id: '5',
            expires: expect.any(Date),
            name: foodName,
            servings: 1
        }));
    });
});
