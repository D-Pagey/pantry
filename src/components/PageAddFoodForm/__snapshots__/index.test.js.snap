// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAddFoodForm component should render 1`] = `
.c1 {
  padding: 0 0 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  background-color: #288efc;
  border: 0;
  border-radius: 0.25rem;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  height: 45px;
  line-height: 1.2;
  width: 200px;
}

.c12 {
  background-color: #288efc;
  border: 0;
  border-radius: 0.25rem;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  height: 45px;
  line-height: 1.2;
  width: 200px;
  background-color: linear-gradient(199.65deg,#288efc 0%,#2863fc 100%);
  border-radius: 6px;
  color: white;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c11 {
  background-color: #288efc;
  border: 0;
  border-radius: 0.25rem;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  height: 45px;
  line-height: 1.2;
  width: 200px;
  background-color: white;
  border-radius: 6px;
  color: black;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c13 {
  background-color: #288efc;
  border: 0;
  border-radius: 0.25rem;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  height: 45px;
  line-height: 1.2;
  width: 200px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 0 1rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(3,max-content);
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;
  margin: 0 0 1rem;
}

.c7 {
  color: blue;
  grid-row: 2;
}

.c6 {
  grid-row: 1;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 0 1rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 0 1rem;
}

.c3 {
  border: 1px solid hsl(0,0%,80%);
  border-radius: 2px;
  padding: 5px;
}

<form
  action="#"
>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="What categories of food?"
    >
      What categories of food?
    </label>
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-5duvv3-control"
      >
        <div
          class=" css-g1d714-ValueContainer"
        >
          <div
            class=" css-1wa3eu0-placeholder"
          >
            Select...
          </div>
          <div
            class="css-b8ldur-Input"
          >
            <div
              class=""
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="What categories of food?"
                spellcheck="false"
                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            class=" css-tlfecz-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-6q0nyr-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c2"
  >
    <label
      class="c1"
      for="addFoodInput"
    >
      What is the name of the food?
    </label>
    <input
      class="c3"
      data-testid="addFoodInput"
      id="addFoodInput"
      name="name"
      placeholder="Name..."
      type="text"
      value=""
    />
    
  </div>
  <div
    class="c4"
  >
    <label
      class="c1"
    >
      When does it expire?
    </label>
    <div
      class="c5"
    >
      <button
        class="c6"
        type="button"
      >
        Up
      </button>
      <span
        class="c7"
        data-testid="dialDatePickerDay"
      >
        24th
      </span>
      <button
        type="button"
      >
        Down
      </button>
      <button
        class="c6"
        type="button"
      >
        Up
      </button>
      <span
        class="c7"
        data-testid="dialDatePickerMonth"
      >
        April
      </span>
      <button
        type="button"
      >
        Down
      </button>
      <button
        class="c6"
        type="button"
      >
        Up
      </button>
      <span
        class="c7"
        data-testid="dialDatePickerYear"
      >
        2020
      </span>
      <button
        type="button"
      >
        Down
      </button>
    </div>
    <button
      class="c8"
      data-testid=""
      type="button"
      width=""
    >
      Add 1 week
    </button>
  </div>
  <div
    class="c9"
    data-testid=""
  >
    <label
      class="c1"
    >
      How many servings?
    </label>
    <div
      class="c10"
    >
      <button
        class="c11"
        data-testid="singleSelectButton0"
        type="button"
        width=""
      >
        1
      </button>
      <button
        class="c12"
        data-testid="singleSelectButton1"
        type="button"
        width=""
      >
        2
      </button>
      <button
        class="c11"
        data-testid="singleSelectButton2"
        type="button"
        width=""
      >
        3
      </button>
      <button
        class="c11"
        data-testid="singleSelectButton3"
        type="button"
        width=""
      >
        4+
      </button>
    </div>
  </div>
  <button
    class="c13"
    data-testid="addFoodFormSubmit"
    type="submit"
    width=""
  >
    Submit
  </button>
</form>
`;
